### ISSUE TITLE
Support for Dreame Bot Z10 Pro Vacuum

### ISSUE THREAD

**Issue by Myth0zz** [merdok/homebridge-miot#62]
Hello, could you add support for this robot vacuum? It's recent. I already have added it in Mi Home, I also have token, IP etc. 

https://www.dreame-technology.com/products/dreame-z10-pro 

Let me know if you need anything else.

**Comment by merdok** [comment]
Hi,

sure I can do that. But could you provide me with the device model? I need the exact device model to properly identify the device and find its features.

**Comment by Myth0zz** [comment]
![23FD092A-AE8D-4128-BA9C-47C78AEE7A99](https://user-images.githubusercontent.com/80846065/126745707-4b8d6e36-0207-4854-a16f-c196326d9a47.jpeg)

of course!! Here it is! Let me know if you need anything else.

**Comment by merdok** [comment]
Unfortunately there i do not see the right device model anywhere. You can check the supported device list to have an idea how the device model should look like.

**Comment by Myth0zz** [comment]
I already asked the manufacturer for a valid device model name. I'm a reviewer, so if all goes well we may have something soon. Please stand by.

P.S. Are you sure the plugin won't work for this device as is? You need this device name 100%?

**Comment by merdok** [comment]
Yes, i need the real device model for two reasons:
1. Lookup the feature set of the device
2. Mark it as supported in the plugin.

I mean you can also add your device in my plugin and enable homebridge debug mode, It will print you the exact model name in the console then, even if it is not yet supported.

**Comment by Myth0zz** [comment]
The Mi Home log file says this: 

dreame.vacuum.p2028 (!) Could this help??

**Comment by Myth0zz** [comment]
I added it in the plugin and my terminal says: 

23/07/2021, 23:33:15 MIoT Device (did: 270868626) is now connected

But I don't see it anywhere in Homekit.

**Comment by merdok** [comment]
Yes, this is now the right device model! Thanks! Will add support for it.

**Comment by Myth0zz** [comment]
OK, that's great. As I mentioned above, my Hoobs terminal says: 

23/07/2021, 23:33:15 MIoT Device (did: 270868626) is now connected

But I don't see anything in Homekit. I guess it will appear when you add support?

**Comment by merdok** [comment]
Yes, i need to add support for your device, it will be added in the next update.

**Comment by Myth0zz** [comment]
Awesome! Waiting ;) Thanks for your instant support.

**Comment by merdok** [comment]
Version 0.9.8 should now support your device!

**Comment by Myth0zz** [comment]
I did the update, however the vacuum doesn't appear in my Homekit.

Terminal says: 25/07/2021, 22:23:52 MIoT Device (did: 270868626) is now connected

Any help?

**Comment by Myth0zz** [comment]
Here's what I have added:

{
    "platforms": [
        {
            "platform": "miot",
            "devices": [
                {
                    "name": "Dreame Z10 Pro",
                    "ip": "192.168.152.100",
                    "token": "355530673xxxxxxxxxxxx2384a79",
                    "pollingInterval": 10,
                    "modeControl": true,
                    "mopModeControl": true,
                    "dndControl": true
                }
            ]
        }
    ]
}

**Comment by merdok** [comment]
your config.json is wrong and cannot be read. You need to fix that.

**Comment by Myth0zz** [comment]
What does this mean? I am not a developer, to understand what you say. Can you please elaborate?

**Comment by merdok** [comment]
You have an error in your config.json and it cannot be read. If you are unsure about what to do then just copy the example config.json from the README.

**Comment by Myth0zz** [comment]
If there was an error in the config.json then it wouldn't be saved and I wouldn't be able to install the plugin, or restart the hoobs. I have added what I wrote 2 messages above if you can kindly check. 

Also, I have 15 devices working great with my config/json all this time. The only thing I added was the code above - I am pasting it here as well:

{
    "platforms": [
        {
            "platform": "miot",
            "devices": [
                {
                    "name": "Dreame Z10 Pro",
                    "ip": "192.168.152.100",
                    "token": "3555306xxxxxxxxxxx2384a79",
                    "pollingInterval": 10,
                    "modeControl": true,
                    "mopModeControl": true,
                    "dndControl": true
                }
            ]
        }
    ]
}

Copying and pasting what's inside the README is not helpful, because it's for a SmartMi Fan and not a vacuum. I would be very much obliged if you could be more clear, because I cannot find where is my mistake. 

Nick

**Comment by merdok** [comment]
I do not see any issue in your config at first sight. That you have other devices working does not matter, since the error is in the particular config above. 
If you are using hoobs then this is a common issue that hoobs does some modifications and breaks the config. How to exactly fix it in hoobs i cannot tell you because i do not use hoobs.

**Comment by Myth0zz** [comment]
Then we did all this for nothing :(

Thank you for your reply. But since I’m not an expert I cannot see what’s wrong. And if I cannot find what’s wrong, I just have a broken plugin.

Nick

Sent from my iPhone

On 25 Jul 2021, at 23:02, Marcin ***@***.***> wrote:

﻿

I do not see any issue in your config at first sight. That you have other devices working does not matter, since the error is in the particular config above.
If you are using hoobs then this is a common issue that hoobs does some modifications and breaks the config. How to exactly fix it in hoobs i cannot tell you because i do not use hoobs.

**Comment by merdok** [comment]
You do not have a broken plugin, but a broken config.json. Someone had already the same issue and provided a fix: https://github.com/merdok/homebridge-webos-tv/issues/239#issuecomment-634760361, if you would try to search then you could find it...

**Comment by Myth0zz** [comment]
I am still trying to find what's wrong. Here's my complete config.json for all my devices:

{
    "server": {
        "port": 80,
        "origin": "*",
        "autostart": 0,
        "home_setup_id": "X-HM://0023ISYWYQLAP",
        "polling_seconds": 5
    },
    "client": {
        "default_route": "status",
        "inactive_logoff": 30,
        "theme": "hoobs-light",
        "locale": "en",
        "temp_units": "celsius",
        "country_code": "GR",
        "postal_code": "26442",
        "latitude": "38.246639",
        "longitude": "21.734573"
    },
    "bridge": {
        "name": "HOOBS",
        "port": 51826,
        "pin": "031-45-154",
        "username": "77:A4:8B:97:59:C9"
    },
    "description": "",
    "ports": {},
    "accessories": [
        {
            "accessory": "SmartmiHeater",
            "name": "Baby Heater",
            "ip": "192.168.152.104",
            "did": "270868626",
            "token": "a00596cfe9f6a578997671309797f7c7",
            "minTemp": 18,
            "maxTemp": 25,
            "enableLED": false,
            "enableLEDName": "LED",
            "enableBuzzer": false,
            "enableBuzzerName": "Buzzer",
            "polling_interval": 60000,
            "plugin_map": {
                "plugin_name": "homebridge-smartmi-heater",
                "index": 0
            }
        }
    ],
    "platforms": [
        {
            "platform": "SensiboAC",
            "plugin_map": {
                "plugin_name": "homebridge-sensibo-ac"
            },
            "disableFan": false,
            "disableDry": false,
            "enableSyncButton": false,
            "syncButtonInAccessory": false,
            "enableOccupancySensor": false,
            "enableClimateReactSwitch": false,
            "enableHistoryStorage": false,
            "disableHorizontalSwing": false,
            "disableLightSwitch": false,
            "externalHumiditySensor": false,
            "allowRepeatedCommands": false,
            "debug": false,
            "username": "mythozz@msn.com",
            "password": "dm@gate21"
        },
        {
            "platform": "SamsungTizen",
            "plugin_map": {
                "plugin_name": "homebridge-samsung-tizen"
            },
            "refresh": {
                "main": true,
                "switch": true
            },
            "devices": [
                {
                    "name": "Samsung TV",
                    "ip": "192.168.152.154",
                    "mac": "CC:6E:A4:5E:12:F3",
                    "uuid": "A1"
                }
            ],
            "name": "SamsungTizen",
            "delay": 400,
            "app_list": true,
            "inputs": [
                {
                    "name": "Netflix",
                    "type": "app",
                    "value": "11101200001"
                },
                {
                    "name": "Prime Video",
                    "type": "app",
                    "value": "3201512006785"
                },
                {
                    "name": "YouTube",
                    "type": "app",
                    "value": "111299001912"
                },
                {
                    "name": "Cosmote TV",
                    "type": "app",
                    "value": "3201805016322"
                },
                {
                    "name": "ERTFLIX",
                    "type": "app",
                    "value": "3202008021483"
                }
            ]
        },
        {
            "platform": "yeelight",
            "plugin_map": {
                "plugin_name": "homebridge-yeelight-wifi"
            }
        },
        {
            "platform": "XboxTv",
            "plugin_map": {
                "plugin_name": "homebridge-xbox-tv"
            },
            "devices": [
                {
                    "name": "Xbox One",
                    "host": "192.168.152.183",
                    "xboxliveid": "FD003BB673EBA4A6",
                    "xboxWebApiToken": "M.R5_BAU.be1c3729-8ae5-d62b-5abd-4323c9c96383",
                    "refreshInterval": 5,
                    "xboxWebApiEnabled": false,
                    "disableLogInfo": false,
                    "volumeControl": 0,
                    "switchInfoMenu": false
                }
            ]
        },
        {
            "platform": "miot",
            "plugin_map": {
                "plugin_name": "homebridge-miot"
            },
            "devices": {
                "platform": "miot",
                "devices": [
                    {
                        "name": "Dreame Z10 Pro",
                        "ip": "192.168.152.100",
                        "token": "35553067333667556531306c32384a79",
                        "pollingInterval": 10,
                        "modeControl": true,
                        "mopModeControl": true,
                        "dndControl": true
                    }
                ]
            }
        }
    ]
}

**Comment by Myth0zz** [comment]
I DID it!

I had to remove the: 
},
"devices": {
"platform": "miot"!

It now appeared!

**Comment by merdok** [comment]
Good to hear. Not sure why hoobs causes this error...

**Comment by Myth0zz** [comment]
Thank you very much for your persistence to solve it. Also thanks for sending me the link to the other error, that helped me locate it.

Keep up the good work.

**Comment by merdok** [comment]
No problem 👍 Glad that in the end you were able to fix!
BTW: Did you already try the vacuum and does it work?

**Comment by Myth0zz** [comment]
Yes just did!

On/off works great. All switches to control mop, do not disturb and suction power work as well!

Excellent so far. I will do more tests tomorrow morning here. It’s 23.30 in the night now :)

Nick

Sent from my iPhone

On 25 Jul 2021, at 23:28, Marcin ***@***.***> wrote:

﻿

No problem 👍 Glad that in the end you were able to fix!
BTW: Did you already try the vacuum and does it work?

**Comment by merdok** [comment]
Perfect :)

**Comment by Myth0zz** [comment]
Merdok hi again! Thank you again for adding my vacuum in Hoobs. I have one question. I recently purchased a Mi Air Fryer BUT it works in Germany servers. I changed server in my Android phone and I see it. I also got token, IP, everything.

But my Whole MI Home (including my Vacuum) is in CN servers. 

CAN I add another server in HOOBS? So that both (CN and DE) can be used simultaneously? 

Is there a way? The plugin now only has option to select 1 server (I have CN).

Waiting on your reply,

Nick
